src/engine/strat_engine/pool.rs

```diff
@@ -59,18 +63,23 @@ impl StratPool {
         let pool_uuid = Uuid::new_v4();
 
         let devices = try!(resolve_devices(paths));
-        let bds = try!(initialize(&pool_uuid, devices, MIN_MDA_SECTORS, force));
-
-        // TODO: We've got some temporary code in BlockDev::initialize that
-        // makes sure we've got at least 2 blockdevs supplied - one for a meta
-        // and one for data.  In the future, we will be able to deal with a
-        // single blockdev.  When that code is added to use a single blockdev,
-        // the check for 2 devs in BlockDev::initialize should be removed.
-        assert!(bds.len() >= 2);
-
-        let meta_dev = try!(LinearDev::new(&format!("stratis_{}_meta", name),
-                                           dm,
-                                           &[bds[0].avail_range_segment()]));
+        let mut block_mgr =
+            BlockDevMgr::new(try!(initialize(&pool_uuid, devices, MIN_MDA_SECTORS, force)));
+
+        if block_mgr.avail_space() < INITIAL_META_SIZE + INITIAL_DATA_SIZE {
+            let short_bytes = *(INITIAL_META_SIZE + INITIAL_DATA_SIZE - block_mgr.avail_space()) *
+                              SECTOR_SIZE as u64;
+            block_mgr.remove_all().expect("cannot de-initialize blockdevs");
```

Comment:
There is no reason to be confident that there is no IO error, so this should be either an unwrap()/TODO or some kind of try!().